include/alpaka/atomic/AtomicOmpBuiltIn.hpp

```diff
@@ -42,7 +42,7 @@ namespace alpaka
 #        pragma omp atomic capture
                 {
                     old = ref;
-                    ref += value;
+                    ref = static_cast<T>(ref + value);
```

Comment:
For some reason, including TBB suddenly generated warnings here, which were turned into errors.

Reply:
I am not sure if a cast within the capture is allowed. It is not defined by the standard. IMO only a binary operator is allowed.
https://www.openmp.org/spec-html/5.0/openmpsu95.html

Reply:
IMO only

```
{ v = x; x binop= expr; } 
{ x binop= expr; v = x; } 
{ v = x; x = x binop expr; } 
{ v = x; x = expr binop x; } 
{ x = x binop expr; v = x; } 
{ x = expr binop x; v = x; } 
{ v = x; x = expr; } 
{ v = x; x++; } 
{ v = x; ++x; } 
{ ++x; v = x; } 
{ x++; v = x; } 
{ v = x; x--; } 
{ v = x; --x; } 
{ --x; v = x; } 
{ x--; v = x; }
```

Is allowed.

Reply:
Replaced by pragmas to disable the warnings.